src/main/java/uk/gov/pay/api/validation/CardLastFourDigitsValidator.java

```diff
@@ -0,0 +1,21 @@
+package uk.gov.pay.api.validation;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.regex.Pattern;
+
+public class CardLastFourDigitsValidator implements ConstraintValidator<ValidCardExpiryDate, String> {
+
+    private Pattern pattern = Pattern.compile("\\d{4}");
+
+    @Override
+    public boolean isValid(String value, ConstraintValidatorContext context) {
+        
+        if (value == null) {
+            return false;
+        }
+
+        return pattern.matcher(value).matches() ? true : false;
+
+    }
+}
```

Comment:
You could have the three pattern based validators extend a common `StringPatternValidator` which would just allow you to set the pattern to be validated, to make this a bit DRYer, and/or a DigitSequenceValidator that takes the number of digits to be validated (not actually sure how to do this off top of my head, but worth exploring)

Reply:
I'm going to leave this for now as I'm happy as it is (and also getting the types to line up was a real pain and it gains little).